src/tests/class_tests/openms/source/ElementDB_test.cpp

```diff
@@ -135,6 +152,70 @@ START_SECTION(void addElement(const std::string& name,
   // ptr addresses cannot change, otherwise we are in trouble since EmpiricalFormula uses those
   TEST_EQUAL(oxygen, new_oxygen)
   TEST_REAL_SIMILAR(oxygen->getAverageWeight(), 16.8994405) // average weight has changed
+
+  TEST_EQUAL(e_ptr->getElement(800), elem_nullPointer)
+  e_ptr->addElement("NewElement", "NE", 800u, oxygen_abundance, oxygen_mass, false);
+  const Element * new_ele = e_ptr->getElement(800);
+  TEST_REAL_SIMILAR(new_ele->getAverageWeight(), 16.8994405) // average weight of new element
+
+  // Test that we cannot add elements twice with replace=false 
+  TEST_EXCEPTION(Exception::IllegalArgument, e_ptr->addElement("NewElement", "NE", 800u, oxygen_abundance, oxygen_mass, false))
+
+  // Test that we can add elements twice with replace=true 
+  e_ptr->addElement("NewElement", "NE", 800u, oxygen_abundance, oxygen_mass, true);
+}
+END_SECTION
+
+START_SECTION( void addIsotope(const std::string& name, const std::string& symbol, const unsigned int an, double abundance, double mass, double half_life, Isotope::DecayMode decay, bool replace_existing))
+{
+  const Isotope * iso1 = e_ptr->getIsotope("(238)U");
+  e_ptr->addIsotope("Uranium", "U", 92u, 1.3, 238.05, 1e5, Isotope::DecayMode::UNKNOWN, true);
+
+  const Isotope * iso2 = e_ptr->getIsotope("(238)U");
+  // ptr addresses cannot change, otherwise we are in trouble since EmpiricalFormula uses those
+  TEST_EQUAL(iso1, iso2)
+  TEST_REAL_SIMILAR(iso1->getAbundance(), 1.3) // natural abundance has changed
+
+  // we have now managed to have 130% natural abundance for Uranium
+  // NOTE: this is a major problem for average weight calculations etc
+  const Element* element = e_ptr->getElement(92);
+  double sum = 0;
+  for (auto& iso : element->getIsotopeDistribution()) {sum += iso.getIntensity();}
+  TEST_REAL_SIMILAR(sum, 1.30725795222315);
+
+  // new Uranium isotope added
+  TEST_EQUAL(e_ptr->getIsotope("(314)C") == nullptr, true)
+  int nr_isotopes = element->getIsotopes().size();
+  e_ptr->addIsotope("Uranium", "U", 92u, 0.3, 314.0, 1e5, Isotope::DecayMode::UNKNOWN, false);
+  const Isotope * new_iso = e_ptr->getIsotope("(314)U");
+  TEST_EQUAL( new_iso != nullptr, true)
+  TEST_EQUAL( new_iso->getElement(), element)
+  TEST_EQUAL( element->getIsotopes().size(), nr_isotopes+1) // increased number of isotopes by one
+  TEST_EQUAL( element->getIsotopeDistribution().getContainer().size(), nr_isotopes+1) // increased number of isotopes by one
+
+  // we have now managed to have 160% natural abundance for Uranium
```

Comment:
maybe this should be checked when updating the isotope distribution?!
Should be 1 +- eps, and exception otherwise?

Reply:
this was always possible in the past, I guess the question is whether we should allow users to do something that does not make sense. One issue could arise, as you update all isotopes with `addIsotope` you could end up in a situation where you are briefly over/below 100%, depending on the order in which you update the isotopes. I also think that any user who messes around with these isotopes (hopefully) knows what they are doing and does their own sanity checks?

Reply:
good point!
I would still argue that the check should be done by us.
What about a static function which creates an isotope, but does not add it to the DB. The user can then create a bunch of those and push them to the DB in one go.